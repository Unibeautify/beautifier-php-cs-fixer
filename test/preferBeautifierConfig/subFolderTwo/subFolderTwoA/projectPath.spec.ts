import * as fs from "fs";
import * as path from "path";

import { newUnibeautify, Beautifier } from "unibeautify";
import beautifier from "@src";
import { raw } from "@test/utils";

test(`should successfully beautify file with config one directory up`, () => {
  const unibeautify = newUnibeautify();
  unibeautify.loadBeautifier(beautifier);
  const text: string = fs
    .readFileSync(path.resolve(__dirname, "../../../fixtures/test4.php"))
    .toString();
  return unibeautify
    .beautify({
      filePath: path.resolve(__dirname),
      projectPath: path.resolve(__dirname, "../"),
      languageName: "PHP",
      options: {
        PHP: {
          "PHP-CS-Fixer": {
            prefer_beautifier_config: true
          }
        } as any,
      },
      text,
    })
    .then(results => {
      expect(raw(results)).toMatchSnapshot();
    });
});
