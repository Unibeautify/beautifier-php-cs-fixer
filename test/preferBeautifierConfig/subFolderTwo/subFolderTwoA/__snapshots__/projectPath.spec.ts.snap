// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should successfully beautify file with config one directory up 1`] = `
<?php
$d = true;
if ($a) {
    return 1;
} else if ($b) {
  return 2;
}
`;
